repository_rules/framework_builder.bzl

```diff
@@ -164,7 +164,16 @@ def _cocoapods_impl(ctx):
     prebuild_subpath = "Pods/_Prebuild/GeneratedFrameworks"
     for path in ctx.path(prebuild_subpath).readdir():
         filegroups = _create_buildfile_content(wd, prebuild_subpath + "/" + path.basename)
-        buildfile_content = buildfile_content + filegroups
+        nested_path = prebuild_subpath + "/" + path.basename + "/Frameworks"
+        nested_filegroups = _create_buildfile_content(wd, nested_path)
+        nested_framework_path = ctx.path(nested_path)
+        if nested_framework_path.exists:
+            for nested_subpath in nested_framework_path.readdir():
+                filegroups = filegroups + _create_buildfile_content(
+                    wd,
+                    prebuild_subpath + "/" + path.basename + "/Frameworks/" + nested_subpath.basename,
+                )
+        buildfile_content = buildfile_content + filegroups + nested_filegroups
```

Comment:
Almost there I think.

Why do we need the `nested_filegroups`? 😅 From what you mentioned I understood that the framework was at `Pods/_Prebuild/GeneratedFrameworks/Google-Mobile-Ads-SDK/Frameworks/GoogleMobileAdsFramework-Current.framework`. Isnt the following enough?:

```starlark
    for path in ctx.path(prebuild_subpath).readdir():
         filegroups = _create_buildfile_content(wd, prebuild_subpath + "/" + path.basename)
         vendored_frameworks_filegroups = _create_buildfile_content(wd, prebuild_subpath + "/" + path.basename + "/Frameworks"")
         buildfile_content = buildfile_content + filegroups + vendored_frameworks_filegroups
```

Reply:
This is now fixed!

Reply:
I cant see any new changes, I keep seeing `nested_filegroups`

> Why do we need the nested_filegroups? 😅

Reply:
I do not see why we need the following:
```
        nested_framework_path = ctx.path(nested_path)
        if nested_framework_path.exists:
            for nested_subpath in nested_framework_path.readdir():
                filegroups = filegroups + _create_buildfile_content(
                    wd,
                    prebuild_subpath + "/" + path.basename + "/Frameworks/" + nested_subpath.basename,
                )
        buildfile_content = buildfile_content + filegroups + nested_filegroups
```

Reply:
The problem is the framework in this case is actually nested below `Frameworks/VendoredPodName-Current`, we need to iterate over that subdirectory, otherwise we get failures in the case the subdirectories do not exist

Reply:
I see. Yes, seems like the vendored framework can be nested at any arbitrary subpath. This solution as is currently in this PR will not solve the problem 😞 Instead, looks like we need to recursively look for `.framework` folders under each of the `GeneratedFrameworks/*` folder. 

One example is `GoogleUserMessagingPlatform`, which has a vendored framework under the path `Frameworks/Release/UserMessagingPlatform.framework`

<img width="1025" alt="Screenshot 2020-10-09 at 00 28 54" src="https://user-images.githubusercontent.com/7938671/95520097-d7956980-09c6-11eb-8d2a-a3016e8bcba5.png">

Reply:
Alright so we need to do this same logic for the next directory level as well then?

Reply:
Or how can we simulate recursion in Starlark?

Reply:
Yes, seems recursion is needed here. Should be no problem to do it in starlark: either its possible to recursively find folders with the `.framework` suffix (not sure if there is a find function in starlark) or you can do a function to recursively visit all folders all folders under a path

Reply:
Sounds good, is there any solid way to tell if a path is a directory or not? I was looking and I could not find a function to do that

Reply:
Not sure from the top of my mind, check the docs

Reply:
This is now fixed, I added logic to test paths to see if they are a directory or not. This works for my sample harness

<img width="284" alt="Screen Shot 2020-10-15 at 11 18 27 AM" src="https://user-images.githubusercontent.com/56700854/96170386-5cf1b000-0ed8-11eb-871f-89bf37317f50.png">

Reply:
But this current solution is not recursively looking for frameworks, as far as I can tell. Its only looking under the following paths:
* `/*.framework`
* `/Frameworks/*.framework`
* `/Frameworks/*/*.framework`

If you want to fix the rule just for your usecase, then this code is perfect. But it should not be merged in this repository, as it does not solve the problem